chore: use prepare instead of create (#101)
superfaceai/docs: docs/guides/map-use-case-to-provider.md

map UsecaseName { // UseCaseName will be different based on the actual use case in the profile
  // Available input fields:
  // `input.field` - required string
  
  // -----------------------

  // 1) Change HTTP method and path to make an HTTP call
  // provider base URL https://swapi.dev/api - base url will be different based on the actual url in provider.json
  http POST "/endpoint/api/{ input.field }" {
    // 2) Specify security scheme id from Provider JSON definition
    security none
  
    // 3) Pass input values to the HTTP request
    request {
      query {
        // param_name = input.fieldName
      }
      headers {}
      body {}
    }
  
    // 4) Map successful HTTP response to the result. The content type is optional
    response 200 "application/json" {
      return map result {
        outputField = "", // optional nulllable string
      }
    
    }
  
    // 5) Optionally map unsuccessful HTTP response to the use case error
    response 500 {
      return map error {
        title = "", // required string
        detail = "", // optional string
      }
    }
  }
}
```

## Map use cases

> Every profile defines one or more _use cases_. You need to map the use case interfaces to the concrete requests and results towards a provider. If you used CLI to bootstrap the map, it will have pre-defined mappings for every profile use case.

Reviewer: ```suggestion
Every profile defines one or more _use cases_. You need to map the use cases to concrete requests to be made to the provider, and specify how to handle their results. If you used the CLI to bootstrap the map, it will have pre-defined stub mappings for every profile use case.
```
Thoughts?

Author: Maybe: `to be made to the provide APIr, and specify how to handle their responses`🤷🏻‍♂️

### Reading use case inputs {#input-object}

Use cases usually define & expect some _inputs_ from the user. These inputs are [defined in profile in the dedicated field](https://superface.ai/docs/comlink/profile#sec-Use-case).

> You can access these inputs via `input` object which is available _inside use case mapping_.  If you used CLI to bootstrap the map, prepared map lists all inputs (eq. `// input.field - required string`) defined by use case. Each input is paired with description containing information about field type, if it is required and can be null.

Reviewer: ```suggestion
You can access these inputs via the `input` object which is available _inside a map_.  If you used the CLI to bootstrap the map, the resulting map lists all inputs (eq. `// input.field - required string`) defined by the use case. Each input is paired with its description, containing information about its type, if it is required, and if it can be null.
```

### Make HTTP request {#http-request}

> Prepared map contains numbered comments guiding you through mapping the use case. Start with specifying the HTTP method and URL path. Provider's default service base URL is provided in comment for your convinience. See [Comlink reference](https://superface.ai/docs/comlink/map#sec-HTTP-Call) for specifying a different service.

Reviewer: ```suggestion
The resulting map contains numbered comments guiding you through mapping of the use case. Start with specifying the HTTP method, and the URL. The provider's default service base URL is provided in the surrounding comments for your convenience. See [Comlink reference](https://superface.ai/docs/comlink/map#sec-HTTP-Call) for specifying a different service.
```

> All widely known HTTP methods are supported. The path is a simple _string_, but can be templated. _E.g. if you need to request resource whose ID was provided in the input, you might want to use something like `/endpoint/api/{ input.field }`._

Reviewer: ```suggestion
All widely known HTTP methods are supported. The path is a simple _string_, but can be templated. _E.g. if you need to request a resource whose ID was provided in the input, you might want to use something like `/endpoint/api/{ input.field }`._
```

Actually, I think I prefer `/resources/{input.id}` - why change it to `/endpoint/api`?

Author: `/endpoint/api` is used in map template. I was thinking that examples in text could use code from templates to give user more context.

Reviewer: Oh! Ok

}
```

> _The above definition makes `POST` HTTP call to [the provider's default service](./add-new-provider.md#default-service) on path `/api/messages`.<br />_

Reviewer: ```suggestion
_The above definition makes a `POST` HTTP call to [the provider's default service](./add-new-provider.md#default-service) on path `/api/messages`.<br />_
```

### Authenticate the request (optional) {#authentication}

> To authenticate the request, simply reference the _security scheme ID_ you want to use for the specific request in `security` definition. Security schemes are defined in Provider JSON documents and prepared map already lists possible security schemes. If provider does not require any type of authentication you can use `scurity none`.

Reviewer: ```suggestion
To authenticate the request, simply reference the _security scheme ID_ you want to use for the specific request. Security schemes are defined in Provider JSON documents, and the prepared map already lists the available security schemes. If the provider does not require any type of authentication, you can use `security none`.
```

### Pass data to request

> You can pass any data to the request by adding `request` block. Inside, you can pass data to _headers, query or body_ by specifiying `headers`, `query` or `body` blocks, respectively. Prepared map by CLI already contains these keyword so you just need to pass data, probably using `input`. 

Reviewer: ```suggestion
You can pass any data to the request by adding a `request` block. Inside, you can pass data to _headers, query or body_ by specifying `headers`, `query` or `body` blocks, respectively. The map prepared by CLI already contains these blocks, so you just need to pass the data, likely using the `input` object. 
```
